---
title: "MFD project and subprojects report"
author: "F. Delogu"
date: "as.Date(now())"
output:
  html_document:
    toc: yes
    toc_depth: 2
    toc_float: yes
    toc_collapsed: yes
  pdf_document:
    toc: yes
    toc_depth: '2'
params:
  debug: no
  debug2: no
  internal: yes
  release: yes
---

<style type="text/css">
body, td {
   font-size: 10pt;
   text-align: justify
}
code.r{
  font-size: 8pt;
}
pre {
  font-size: 8pt
}
</style>

# Setup

## Install libraries

Installing libraries not included in the mamba environment.
```{r install libraries, message=F, warning=F, echo=params$debug}
if(!require("devtools")) install.packages("devtools")
devtools::install_github("sebastianbarfort/mapDK")
```

## Load libraries
```{r load libraries, message=F, warning=F, echo=params$debug}
library(tidyverse)
library(knitr)
library(ghql)
library(jsonlite)
library(lubridate)
library(mapDK)
library(rgdal)
library(sp)
library(mapproj)
library(openxlsx)
library(terra)

options(width = 500)
```

## Set the environment
```{r setup, echo=params$debug}
options(stringsAsFactors = F, gsubfn.engine = "R")

Sys.setenv("LANGUAGE"="En")

wd <- getwd()

ontology.path <- paste0(wd, '/../../data/ontology/2023-08-28_mfd-habitat-ontology.xlsx')
downloads.path <- paste0(wd, '/../../data/downloads')
metadata.path <- paste0(wd, '/../../data/metadata')

mfd_projects.path <- paste0(wd, '/../../data/results/mfd_projects.xlsx')
minimal_indices_explanation <- paste0(metadata.path, '/general/minimal_metadata_descriptors.xlsx')
reformat.metadata.path <- paste0(wd, '/../../analysis/metadata')
mails_01.path <- paste0(wd, '/../../analysis/mails_01')

releases.path <- paste0(wd, '/../../analysis/releases')

minimal_indices <- c("fieldsample_barcode", "sample_type", "sampling_date",
                     "latitude", "longitude",  "habitat_type", "habitat_typenumber", "sitename",
                     "mfd_sampletype", "mfd_areatype", "mfd_hab1", "mfd_hab2", "mfd_hab3") # NA is interpreted as all

release_indices <- c("project_id",
                     "fieldsample_barcode", "sample_type", "sampling_date",
                     "latitude", "longitude",  "habitat_type", "habitat_typenumber", "sitename",
                     "mfd_sampletype", "mfd_areatype", "mfd_hab1", "mfd_hab2", "mfd_hab3") # NA is interpreted as all

eval_debug <- params$debug
eval_debug2 <- params$debug2
eval_internal <- params$internal
eval_release <- params$release
```

This section is meant for internal use and debugging, if you are interest in seeing the code please knit again the document setting the "debug" flag to "true". Optionally, set "debug2" as "true" to see also the code to generate the debug headings and text.

For the generation of this document the parameters were set as:
```{r print params, echo=eval_debug}
data.frame(param=c("debug", "debug2", "internal", "release"), value=c(eval_debug, eval_debug2, eval_internal, eval_release)) %>%
  kable()
```

## Custom functions

### Load functions
```{r load custom functions, message=F, warning=F, echo=eval_debug}
source(paste0(wd, "/mfd_report_test_routine.R"))
```

### Print functions
```{r print custom functions, message=F, echo=eval_debug, eval=eval_debug}
methods(pstats)
```

## Load data

### Download metadata from online Microflora Danica database
This is the database that is linked to initial data-entry via the Microflora Danica App.
```{r get minimal metadata, echo=eval_debug}
token <- Sys.getenv("GITHUB_GRAPHQL_TOKEN")

con <- GraphqlClient$new(url = "https://mfd.programming.cool/graphql/",
                         headers = list(Authorization = paste0("Bearer ", token)))

qry <- Query$new()

qry$query('fieldSamplesquery', 
          '{fieldsamples {
                barcode
                type
                gpslocation
                habitattype
                habitattypenumber
                sitename
                received
                scanned
                samplingcomment
                project {id}}}')

executedQuery <- con$exec(qry$queries$fieldSamplesquery)

mfd_db0 <- executedQuery %>%
  fromJSON(simplifyVector = TRUE, flatten = TRUE) %>%
  as.data.frame()

colnames(mfd_db0) <- c("fieldsample_barcode", "sample_type", "gps_location", "habitat_type",
                      "habitat_typenumber", "sitename", "received", "scanned",
                      "sampling_comment", "project_id")

rm(con, qry, executedQuery, token)
```

### Time lag check
```{r time lag check, echo=eval_debug}
tmp_diff <- mfd_db0 %>%
  mutate(received = dmy_hm(received),
         scanned = dmy_hm(scanned),
         time_diff = difftime(received, scanned, units = "mins")) %>%
  filter(hour(scanned)!=0,
         hour(received)!=0) %>%
  filter(time_diff>0) %>% 
  arrange(time_diff) %>%
  select(fieldsample_barcode,
         scanned,
         received,
         time_diff,
         project_id)

table(tmp_diff$project_id)

a <- tmp_diff %>%
  #filter(time_diff<5000) %>% 
  pull(time_diff)

hist(as.numeric(a[!is.na(a)]), main="Received-scanned > 0 (mins)")
rm(tmp, a)
```

### Clean up general problems in the database metadata
```{r curate mfd project, echo=eval_debug}
mfd_db <- mfd_db0 %>%
  rowwise() %>%
  mutate(received = dmy_hms(received, truncated = 3) %>% format("%Y-%m-%d")) %>%        # format dates
  mutate(scanned = dmy_hms(scanned, truncated = 3) %>% format("%Y-%m-%d")) %>%        # format dates
  mutate(received=min(c(scanned, received), na.rm = T)) %>%                              # assume that the earliest date between received and scanned is the sampling date, if one is NA the other one is selected
  ungroup() %>%
  select(-scanned) %>%
  #filter(str_detect(fieldsample_barcode, "^MFD")) %>%                                   # not filtering for strange (unmatched) barcodes
  filter(!str_detect(sampling_comment, "extraction neg") | is.na(sampling_comment)) %>% # remove ext negative
  mutate(across(!c(fieldsample_barcode, project_id), ~tolower(as.character(.)))) %>%    # make all lowercase
  mutate(across(habitat_typenumber, ~ toupper(.))) %>%                                  # format habitat number
  mutate(across(habitat_typenumber, ~ str_replace_all(., "X", "0"))) %>%
  separate(gps_location, into = c("latitude", "longitude"), sep = ",") %>%              # split gps
  mutate(latitude = as.numeric(latitude),
         longitude = as.numeric(longitude)) %>%
  mutate(mfd_sampletype = NA,                                                           # Add classifications
         mfd_areatype   = NA,
         mfd_hab1       = NA,
         mfd_hab2       = NA,
         mfd_hab3       = NA,
         sampling_date  = NA) %>%
  as.data.frame() %>%
  mutate(project_id = if_else(fieldsample_barcode %in% c("MFD03760", "MFD03795", "MFD03805",
                                                         "MFD03808", "MFD03846", "MFD03855",
                                                         "MFD03867", "MFD03871", "MFD03908",
                                                         "MFD03970", "MFD03973"),
                              "P08_3", project_id))

rm(mfd_db0)
```

### Format mfd dataframe
```{r format mfd projects, echo=eval_debug}
mfd_projects <- tibble(project_id = unique(c(mfd_db$project_id,
                                             "P04_8", "P09_3",
                                             "P09_4", "P12_5",
                                             "P18_2", "P20_1",
                                             "P21_1", "P13_3",
                                             "P08_8")) %>% sort(), 
                       project_name = NA,
                       description = NA,
                       extended_metadata = NA,
                       people = NA,
                       responsible = NA,
                       comment = NA,
                       metadata_map = NA)
```

### Load the map of Denmark
```{r load DK map, message=F, warning=F, error=F, echo=eval_debug}
if(!file.exists(paste0(downloads.path,"/DK_regions.geojson"))){
  url = "https://api.dataforsyningen.dk/landsdele?format=geojson"
  geofile = tempfile()
  download.file(url, geofile)
  regions <- rgdal::readOGR(geofile)
  rgdal::writeOGR(regions, paste0(downloads.pathh, "/DK_regions.geojson"), layer="file76fc3f3c3b00", driver="GeoJSON")
} else {
  regions <- rgdal::readOGR(paste0(downloads.path, "/DK_regions.geojson"))
}
```

# Subproject curation

## P01_1: Biowide - Original
```{r P01_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P01_1"
pnm <- "Biowide - Original"
des <- "Biowide (Biodiversity in Width and Depth). In light of the biodiversity crisis and our limited ability to explain variation in biodiversity, tools to quantify spatial and temporal variation in biodiversity and its underlying drivers are critically needed. Inspired by the recently published ecospace framework, we developed and tested a sampling design for environmental and biotic mapping. We selected 130 study sites (40×40 m) across Denmark using stratified random sampling along the major environmental gradients underlying biotic variation. Using standardized methods, we collected site species data on vascular plants, bryophytes, macrofungi, lichens, gastropods and arthropods. To evaluate sampling efficiency, we calculated regional coverage (relative to the known species number per taxonomic group), and site scale coverage (i.e., sample completeness per taxonomic group at each site). To extend taxonomic coverage to organisms that are difficult to sample by classical inventories (e.g., nematodes and non-fruiting fungi), we collected soil for metabarcoding. Finally, to assess site conditions, we mapped abiotic conditions, biotic resources and habitat continuity. The main paper is described here: https://bmcecol.biomedcentral.com/articles/10.1186/s12898-019-0260-x. See a list of related publications here: https://ecos.au.dk/forskningraadgivning/temasider/publikationer."
ext <- "Environmental parameters; alternative habitat classifications; plant counts; animal counts; fungal counts; ITS2 amplicons; trnL amplicons."
peo <- "Tobias Guldberg Frøslev <tfroeslev@gbif.org>; Rasmus Ejrnæs <rasmus@bios.au.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")
mapping.project <- readxl::read_xlsx(paste0(metadata.path, "/", pid, "/MFDP005_metadata_32.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame(paste0(metadata.path, "/", pid, "/P01_1_filled_table_TGF.xlsx"))

filled.table1 <- read.xlsx(paste0(metadata.path, "/", pid, "/P01_1_filled_table_TGF.xlsx"), sheet = 1, detectDates = T) %>%
                            mutate(correct_date = sampling_date) %>%
  select(fieldsample_barcode, correct_date)

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table2 <- read.xlsx(paste0(metadata.path, "/", pid, "/P01_1_filled_table.xlsx")) %>%
  mutate(correct_habitat_typenumber = as.character(habitat_typenumber)) %>%
  select(fieldsample_barcode,
         correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid & habitat_type == "natural_soil", "Natural", mfd_areatype),
         mfd_areatype       = ifelse(project_id == pid & habitat_type == "agriculture", "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date),
         sampling_comment   = ifelse(project_id == pid & sampling_comment == "", NA, sampling_comment)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  left_join(filled.table1, by = "fieldsample_barcode") %>%
  mutate(sampling_date      = if_else(project_id == pid, correct_date, sampling_date)) %>%
  select(-correct_date) %>%
  left_join(filled.table2,
            by="fieldsample_barcode") %>%
  mutate(habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_habitat_typenumber)

paragraph2 <- paste0("Considering your previously provided data, we were wondering whether the sites corresponing to \"plantation forests\" can be assigned to habitat_typenmubers within the 9xxx categories (\"Forests\"), to fit with the rest of the data. If this is the case, could you please fill the most specific habitat_typenumber in the highlighted cells?\n\n",
                     "In the file you sent us (\"FinalVægteTilUniquenessUdregning.xlsx\"), we have a few questions. For the cases of sitenames \"Knagerne\" and \"Urskoven\" should we choose 9110 or 9120? For the cases where we have \"pilekrat, §3mose\" and \"§3 mose, rørsump/højstaude\" can we please get the corresponding most specific habitat_typenumber?\n",
                     "For the samples marked with \"Agriculture\" as \"habitat_type\", do you have information on the crop type or can you chose from the attached ontology file?")

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
rm(filled.table1, filled.table2)
```

## P01_2: Biowide - River valleys
```{r P01_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P01_2"
pnm <- "Biowide - River valleys"
des <- "Biowide extension with samples from river valleys"
ext <- "Environmental parameters; alternative habitat classifications; plant counts; animal counts; fungal counts"
peo <- "Tobias Guldberg Frøslev <tfroeslev@gbif.org>; Rasmus Ejrnæs <rasmus@bios.au.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>; Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

date_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P01_2_MFD metadata_32.xlsx")) %>%
  select(fieldsample_barcode = sample_id, correct_date = Sampling_date) %>%
  mutate(correct_date = dmy(correct_date) %>% format("%Y-%m-%d"))

mapping.project <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P01_2_MFD metadata_32.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P01_2_filled_table.xlsx"), sheet = 1, detectDates = T) %>%
                            mutate(correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode, correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  left_join(date_correction,
            by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         sampling_date      = ifelse(fieldsample_barcode == "MFD00449", "2018-08-01", sampling_date)) %>% # Assuming it was taken at the same time
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(habitat_typenumber = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "6402", "6411", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "6403", "6431", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "7007", "7700", habitat_typenumber)) %>%
  select(-correct_date,
         -correct_habitat_typenumber) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P02_1: Biowide - Ecospace
```{r P02_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P02_1"
pnm <- "Biowide - Ecospace"
des <- "Biowide Ecospace. Not sure what the aim of this study was."
ext <- "Environmental parameters; alternative habitat classifications; plant counts; animal counts; fungal counts"
peo <- "Rasmus Ejrnæs <rasmus@bios.au.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>; Per Halkjær Nielsen <phn@bio.aau.dk>; Albert Vejlin Stefansen <a.stefansen@rn.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

data_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P02_1_MFD metadata_32.xlsx")) %>%
  select(fieldsample_barcode = sample_id, correct_date = Sampling_date) %>%
  mutate(correct_date = dmy(correct_date) %>% format("%Y-%m-%d"))

mapping.project <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P02_1_MFD metadata_32.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()
  
filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P02_1_filled_table.xlsx")) %>%
  mutate(correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode,
         correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  left_join(data_correction, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid & habitat_type == "natural_soil", "Natural", mfd_areatype),
         mfd_areatype       = ifelse(project_id == pid & habitat_type == "agriculture", "Agriculture", mfd_areatype),
         mfd_areatype       = ifelse(project_id == pid & str_detect(sampling_comment, "roadverge"), "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid &  str_detect(sampling_comment, "roadverge"), "6100", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, correct_date, sampling_date)) %>%
  select(-correct_date) %>%
  left_join(filled.table) %>%
  mutate(habitat_typenumber = ifelse(project_id == pid & is.na(habitat_typenumber), correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_habitat_typenumber) %>%
  mutate(mfd_areatype = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Agriculture" & habitat_typenumber == "5100", "Natural", mfd_areatype),
         mfd_areatype = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Agriculture" & habitat_typenumber == "6403", "Natural", mfd_areatype),
         mfd_areatype = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Agriculture" & habitat_typenumber == "6010", "Natural", mfd_areatype),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "6402", "6411", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "6403", "6431", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path)

paragraph2 <- paste0("For the samples marked with \"Agriculture\" as \"habitat_type\", do you have information on the crop type or can you chose from the attached ontology file?\n",
                     "Moreover, could you please provide feedback on the 6100 habitat_typenumber? E.g., do you think we should change it to something else or be more specific - for insgance make a 6110 number (corresponding to hedgerow) and 6120 (corresponding to readverge) or others?")

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P02_2: Biowide - Agriculture
```{r P02_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P02_2"
pnm <- "Biowide - Agriculture"
des <- "Thes samples are taken in the Biowide design (40x40 meter (or a narrower and longer grid), 81 cores pooled and subsampled). They were taken to extend agricultural sites, with the specific aim of testing possible effect of three different agricultural practices."
ext <- ""
peo <- "Tobias Guldberg Frøslev < tobiasgf@sund.ku.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>; Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""
metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

data_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P02_2_MFD metadata_32.xlsx")) %>%
  select(fieldsample_barcode = sample_id, correct_date = Sampling_date) %>%
  mutate(correct_date = dmy(correct_date) %>% format("%Y-%m-%d"))

mapping.project <- rbind(readxl::read_excel(paste0(metadata.path, "/", pid, "/P02_2_MFD metadata_32.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame(),
  data.frame(sample_name="F8.1_K", sample_id="MFD00398B"))

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P02_2_filled_table.xlsx"), sheet = 1)  %>%
  mutate(correct_sitename = sitename) %>%
  select(fieldsample_barcode,
         correct_sitename)

mfd_db <- mfd_db %>%
  left_join(data_correction, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "31", "6612", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "22", "6713", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "11", "6123", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "1", "6141", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         sampling_date      = ifelse(fieldsample_barcode == "MFD00398B", "2018-12-04", sampling_date)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_date) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(sitename = if_else(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_sitename)

paragraph2 <- "Could you please verify the GPS coordinates? It looks like there are pairs of samples with the exact coordinates whilst they were supposedly sampled from the edge and the center of the same field."

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P03_1: Biowide - Urban
```{r P03_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P03_1"
pnm <- "Biowide - Urban"
des <- "Related the soil tracker project, see \"Treated like dirt: Robust forensic and ecological inferences from soil eDNA after challenging sample storage\" https://onlinelibrary.wiley.com/doi/10.1002/edn3.367"
ext <- ""
peo <- "Tobias Guldberg Frøslev <tfroeslev@gbif.org>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

data_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P03_1_MFD metadata_32.xlsx")) %>%
  select(fieldsample_barcode = sample_id, correct_date = Sampling_date) %>%
  mutate(correct_date = dmy(correct_date) %>% format("%Y-%m-%d"))

mapping.project <- rbind(readxl::read_excel(paste0(metadata.path, "/", pid, "/P03_1_MFD metadata_32.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame(),
  data.frame(sample_name="PD001", sample_id="MFD00457B"))

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P03_1_filled_table.xlsx"), sheet = 1) %>%
  mutate(correct_latitude = as.numeric(latitude),
         correct_longitude = as.numeric(longitude)) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude)

mfd_db <- mfd_db %>%
  left_join(data_correction, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid & sample_type == "soil", "Soil", mfd_sampletype),
         mfd_sampletype     = ifelse(project_id == pid & sample_type == "sediment", "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "city_parks", "6410", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "city_other", "6420", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "city_pond", "3110", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         sampling_date=ifelse(fieldsample_barcode=="MFD00457B", "2017-11-02", sampling_date)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_date) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(latitude          = if_else(project_id == pid, correct_latitude, latitude),
         longitude         = if_else(project_id == pid, correct_longitude, longitude)) %>%
  select(-correct_latitude,
         -correct_longitude)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_1: Agriculture - Potato
```{r P04_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_1"
pnm <- "Agriculture - Potato"
des <- "Samples from potato fields looking at potato diseases. Not possible to get gps coordinates or farmer agreements."
ext <- ""
peo <- "Kåre Lehman Nielsen <kln@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Not to be included in MfD"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mapping.project <- readxl::read_excel(paste0(metadata.path, "/", pid, "/MFDP006_MFD metadata.xlsx")) %>%
  filter(!is.na(sample_id)) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid , "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "6429", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_2: Agriculture - Aaskov
```{r P04_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_2"
pnm <- "Agriculture - Aaskov"
des <- "Samples from the Aaskov research station with the aim of looking at soil nitrification over two seasons."
ext <- ""
peo <- "Henning Carlo Thomsen <henning.thomsen@agro.au.dk>; Bent Tolstrup Christensen <bent.t.christensen@agro.au.dk>; Albert Vejlin Stefansen <albert_stefansen@hotmail.com>; Andrew Giguere <andrew.giguere@univie.ac.at>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P04_2_filled_table.xlsx")) %>%
  mutate(correct_date = ymd(sampling_date) %>% format("%Y-%m-%d"),
         correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode,
         correct_date,
         correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date),
         habitat_typenumber = ifelse(project_id == pid, NA, habitat_typenumber)) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(sampling_date       = ifelse(project_id == pid, correct_date, sampling_date),
         habitat_typenumber = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_date,
         -correct_habitat_typenumber)

paragraph2 <- paste0("Could you please provide the grid position as the sitename?\n",
                     "For all samples, do you have information on the crop type or can you chose from the attached ontology file?")

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_3: Agriculture - Kvadratnet
```{r P04_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_3"
pnm <- "Agriculture - Kvadratnet"
des <- "Kvadratnet for nitrate levels – yearly investigation of selected number of kvadratnet stations. Stations have been investigated since the 1980’s. Samples included for 2020 and 2021."
ext <- ""
peo <- "Camilla Lemming <cal@seges.dk>; Rita Hørfarter <rih@seges.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "OBS: Date is estimated from a report that mentions that samples are taken in february. https://www.landbrugsinfo.dk/public/f/2/f/godskning_kvalstofprognosen_2018. Publications: So far unclear if we should make dedicated joint publication on this dataset. The sampling occurred over 12 days (3-14 Feb in 2020 and 1-12 Feb 2021) and in case the sampling date was missing we used the middle of the sampling period (8 and 6 Feb for 2020 and 2021 respectively). 14 samples have been confirmed coming from not carachterised fields. In two cases from the 2020 sampling campaing the reported year was 2018 and corrected to 2020."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

hab_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/2022-11-28_mfd-metadata-kvnet.xlsx")) %>%
   separate(unified, into = c("a", "b", "kvnet_hab"), sep = "_") %>%
  select(-a, -b)

mapping.project <- inner_join((readxl::read_excel(paste0(metadata.path, "/", pid, "/mapping.xlsx")) %>%
  filter(!is.na(MFD_id)) %>%
  filter(`Kvnet Punktnr`!="Ikke Kvnet") %>%
  select(`Kvnet Punktnr`, MFD_id) %>%
  as.data.frame()),
  (mfd_db %>%
     mutate(Year = year(received)) %>%
     select(fieldsample_barcode, Year)),
  by=c("MFD_id"="fieldsample_barcode")) %>%
  mutate(sample_id = paste0(`Kvnet Punktnr`, " ", Year)) %>%
  select(sample_id, MFD_id)

filled.table <- rbind((read.xlsx(paste0(metadata.path, "/", pid, "/Kvnet_prøvedatoer 2020.xlsx"), sheet = 1) %>%
                         mutate(Year = 2020)),
                      (read.xlsx(paste0(metadata.path, "/", pid, "/Kvnet_prøvedatoer 2021.xlsx"), sheet = 3) %>%
                         mutate(Year = 2021))) %>%
  mutate(Dato = ymd(Dato) %>% format("%Y-%m-%d"),
         Proevenr = paste0(Proevenr, " ", Year)) %>%
  select(Proevenr, Dato) %>%
  distinct(Proevenr, .keep_all = T) # in case of multiple sampling in the same place in the same year only the earliest is kept

mfd_db <- mfd_db %>%
  left_join(hab_correction, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid & habitat_type == "agriculture", "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = case_when(project_id == pid & kvnet_hab != "6099" ~ kvnet_hab,
                                        project_id == pid & kvnet_hab == "6099" ~ "6000",
                                        .default = habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-kvnet_hab) %>%
  left_join((mapping.project %>%
               separate(sample_id, into=c("location", "Year"), sep=" ", remove=F) %>%
               select(-Year)),
            by = c("fieldsample_barcode"="MFD_id")) %>%
  mutate(sitename           = if_else(project_id == pid, location, sitename)) %>%
  select(-location) %>%
  left_join(filled.table,
            by=c("sample_id"="Proevenr")) %>%
  mutate(sampling_date = case_when( project_id == pid & !is.na(Dato) ~ Dato,
                                    project_id == pid & is.na(Dato) & year(received) == 2020 ~ "2020-02-08",
                                    project_id == pid & is.na(Dato) & year(received) == 2021 ~ "2021-02-06",
                                    fieldsample_barcode == "MFD04074" ~ "2020-02-07",
                                    fieldsample_barcode == "MFD04123" ~ "2020-03-11",
                                    .default = sampling_date)) %>%
  select(-sample_id,
         -Dato)

paragraph2 <- "Were the pairs of samples, sampled the same day, exactly one year apart (as it looks like from the sampling_date column)?"

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_4: Agriculture - Students
```{r P04_4, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_4"
pnm <- "Agriculture - Students"
des <- "Student project aimed to analyze conventional vs. ecological agriculture soil. 7 sem. students."
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

unrecoverable.samples <- c("MFD04003", # We'll not retrieve more info from those samples
                           "MFD04004",
                           "MFD04014",
                           "MFD04020",
                           "MFD04032",
                           "MFD04044",
                           "MFD04056",
                           "MFD04069",
                           "MFD04081",
                           "MFD04105",
                           "MFD04128")

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P04_4_filled_table.xlsx")) %>%
  filter(!fieldsample_barcode %in% unrecoverable.samples) %>%
  mutate(correct_habitat_typenumber = habitat_typenumber,
         correct_sitename = sitename) %>%
  select(fieldsample_barcode,
         correct_habitat_typenumber,
         correct_sitename)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date),
         sampling_date      = ifelse(project_id == pid & str_detect(sampling_comment, "kvadratnet"), NA, sampling_date),
         ) %>%
  filter(!fieldsample_barcode %in% unrecoverable.samples) %>%
  left_join(filled.table) %>%
  mutate(habitat_typenumber = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         sitename           = ifelse(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_habitat_typenumber,
         -correct_sitename) %>%
  add_ontology(ontology_path = ontology.path)

paragraph2 <- paste0("Could you please provide the grid position as the sitename?\n",
                     "For all samples, do you have information on the crop type or can you chose from the attached ontology file?")

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
rm(unrecoverable.samples)
```

## P04_5: Agriculture - Seges
```{r P04_5, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_5"
pnm <- "Agriculture - Seges"
des <- "Agricultural soil collected by farms after harvest, dried and used for normal soil parameter analyses prior to MfD. OK-Lab have made soil-chemistry analyses."
ext <- "Soil parameters and historic crop data."
peo <- "Camilla Lemming <cal@seges.dk>; Rita Hørfarter <rih@seges.dk>; Ole Kristensen <oklab@mail.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "4 Samples were collected in the wrong fields and are filtered out (MFD07683, MFD07246, MFD09066, MFD08545)."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

hab_correction <- rbind(readxl::read_excel(paste0(metadata.path, "/", pid, "/2022-11-28_mfd-metadata-seges.xlsx")) %>%
  separate(unified, into = c("a", "b", "seges_hab"), sep = "_") %>%
  select(fieldsample_barcode, seges_hab),
  data.frame(fieldsample_barcode=c("MFD08223", "MFD07450"),
             seges_hab=c("6822", "6822")))

to_remove <- c("MFD07683",
               "MFD07246",
               "MFD09066",
               "MFD08545")

mapping.project <- readxl::read_excel(paste0(metadata.path, "/", pid, "/2022-11-28_mfd-metadata-seges.xlsx"),
                                     sheet = "MFD to SEGES") %>%
  filter(!is.na(bægre)) %>%
  mutate(sample_name=bægre, sample_id=mfd) %>%
  select(sample_name, sample_id) %>%
  as.data.frame() %>%
  filter(!sample_id %in% to_remove)

mfd_db <- mfd_db %>%
  left_join(hab_correction,
            by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid & habitat_type == "agriculture", "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, seges_hab, habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & is.na(habitat_typenumber), "6000", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-seges_hab) %>%
  filter(!fieldsample_barcode %in% to_remove) %>%
  mutate(latitude           = ifelse(project_id == pid, NA, latitude),
         longitude          = ifelse(project_id == pid, NA, longitude))

paragraph2 <- "As you can see from the table, we are missing all the info from sample MFD07450. Could you please send us the list of your sample IDs so that we can link our data and find the missing sample."

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_6: Agriculture - SINKS
```{r P04_6, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_6"
pnm <- "Agriculture - SINKS"
des <- "Investigate peatlands (areas with waterlogged conditions), which prevent plant material being fully decomposed. In addition to the 150 agriculture peatlands samples 150 samples from same area but of natural sites are being collected Sampling during 6 weeks period summer 2021."
ext <- ""
peo <- "Lis Wollesen de Jonge <lis.w.de.jonge@agro.au.dk>; Anne-Cathrine Storgaard Danielsen <acsd@agro.au.dk>; Cecilie Hermansen <cecilie.hermansen@agro.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "græs"), "6200", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "korn"), "6100", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "brak"), "6900", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "raps"), "6710", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & is.na(habitat_typenumber), "6000", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, "2021-07-01", sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P04_6_filled_table.xlsx")) %>%
  mutate(correct_sitename = sitename) %>%
  select(fieldsample_barcode, correct_sitename)

mfd_db <- left_join(mfd_db, filled.table,
                 by="fieldsample_barcode") %>%
  mutate(sitename      = ifelse(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_sitename)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_7: Agriculture - Grazing
```{r P04_7, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_7"
pnm <- "Agriculture - Grazing"
des <- "Investigate soil samples from 3 farmers about effect of organically grown crops. See: https://www.facebook.com/SundJordSundKost/"
ext <- ""
peo <- "Jørgen Friis Pedersen <friisnibe@gmail.com>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P04_7_filled_table.xlsx")) %>%
  separate(habitat_typenumber, into=c("f1", "f2", "habitat_typenumber")) %>%
  mutate(correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode, correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Agriculture", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(habitat_typenumber = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_habitat_typenumber) %>%
  add_ontology(ontology_path = ontology.path)

paragraph2 <- "Could you please provide us with a habitat_typenumber per sample that matches the ones on the attached ontology (or suggest new ones follwoing the same scheme)."

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P04_8: Agriculture - LUCAS
```{r P04_8, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P04_8"
pnm <- "Agriculture - LUCAS"
des <- "LUCAS: Land Use and Coverage Area frame Survey, a project started by EUROPEAN SOIL DATA CENTRE (ESDAC). Project homepage: https://esdac.jrc.ec.europa.eu/projects/lucas."
ext <- ""
peo <- "Lis Wollesen de Jonge <lis.w.de.jonge@agro.au.dk>; Anne-Cathrine Storgaard Danielsen <acsd@agro.au.dk>; Cecilie Hermansen <cecilie.hermansen@agro.au.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "Previous created as P26_1 Natural/Agriculture? – Greenland 123 samples supplied by Foulum – dried soil"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P04_8_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
    sampling_date = sampling_date  %>% format("%Y-%m-%d"),
    received = NA,
    mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
    mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper)) %>%
  #select(-project_name) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])
  #full_join(filled.table)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P05_1: Urban - Rainwater
```{r P05_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P05_1"
pnm <- "Urban - Rainwater"
des <- "Investigate microorganisms and heavy metals in rainwater basins across Denmark (highway and city rainwater basins). NB: samples with sediment core – to avoid metal interference for heavy metal measurements."
ext <- ""
peo <- "Jes Vollertsen <jesvollertsen@build.aau.dk>; Dian Stephansen <dast@build.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Marta Anna Nierychlo <mni@bio.aau.dk>"
com <- "In theory the dried sampels are not sediment, but soil samples. Double check with the ontology scheme."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid , "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & str_detect(habitat_type, "rainwater_city"), "3120", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(habitat_type, "highway"), "3130", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "dried"), "3131", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "dried"), "3131", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "dry"), "3131", habitat_typenumber),         
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P05_2: Urban - Ponds
```{r P05_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P05_2"
pnm <- "Urban - Ponds"
des <- "The goal was to get samples to MfD from \"urban\" waters, they may have another community composition compared to \"natural\" waters (lakes and streams). Also to look for potential differences across Denmark. Sediment samples taken from towns/cities: small and very small lakes (branddamme)."
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "The responisble people are Erika Dvarionaite and Amalie Berg-Mortensen (email to be found). Could perhaps go together with data from all MfD lakes (P09_1)."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "3110", habitat_typenumber),
         sampling_comment   = ifelse(is.na(sampling_comment), "", sampling_comment),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "dried"), "3111", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P06_1: Subterranean - Polluted
```{r P06_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P06_1"
pnm <- "Subterranean - Polluted"
des <- "Part of \"Biomap\" project at AU (to develop the use of electrical fields to trace soil pollutants): https://pure.au.dk/portal/en/projects/biomap--microbiologically-assisted-mapping-of-soil-pollution(136bad80-1ab8-4234-abfd-d3ae63384933).html. Viby: oil contamination; Ringe: asfalt contamination; Himmark: Chlorinated hydrocarbons contam. 1-3 meters. Controls: 1 m"
ext <- ""
peo <- "Andreas Schramm <andreas.schramm@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "No GPS on samples, estimated from sampling_comment where a city is mentioned."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid  & str_detect(sampling_comment, "pollut"), "6130", habitat_typenumber),         
         habitat_typenumber = ifelse(project_id == pid  & str_detect(sampling_comment, "control"), "6000", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid  & str_detect(sampling_comment, "trace"), "6120", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid  & str_detect(sampling_comment, "below detection"), "6190", habitat_typenumber),
         latitude = ifelse(project_id == pid & str_detect(sampling_comment, "viby"),56.121,  latitude),
         longitude = ifelse(project_id == pid & str_detect(sampling_comment, "viby"),10.146,  longitude),
         latitude = ifelse(project_id == pid & str_detect(sampling_comment, "ringe"),55.223,  latitude),
         longitude = ifelse(project_id == pid & str_detect(sampling_comment, "ringe"),10.480,  longitude),    
         latitude = ifelse(project_id == pid & str_detect(sampling_comment, "himmark"),55.043,  latitude),
         longitude = ifelse(project_id == pid & str_detect(sampling_comment, "himmark"),9.852,  longitude),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P06_2: Subterranean – Polluted soil, city
```{r P06_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P06_2"
pnm <- "Subterranean – Polluted soil, city"
des <- "Deep soil test samples"
ext <- ""
peo <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Vibike Rudkjøbing Jørgensen <vrj@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table1 <- read.xlsx(paste0(metadata.path, "/", pid, "/P06_2_frde_standard table.xlsx"), sheet=1) %>%
  filter(fieldsample_barcode%in%mfd_db$fieldsample_barcode) %>%
  mutate(correct_latitude = latitude,
         correct_longitude = longitude,
         correct_sitename = sitename,
         correct_habitat_typenumber = habitat_typenumber,
         correct_comment = comment) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude,
         correct_sitename,
         correct_habitat_typenumber,
         correct_comment)

filled.table2 <- (read.xlsx(paste0(metadata.path, "/", pid, "/P06_2_frde_standard table.xlsx"), sheet=1) %>%
  filter(!fieldsample_barcode%in%mfd_db$fieldsample_barcode)%>%
  mutate(sampling_comment = comment,
         received = NA,
         project_id = pid) %>%
    select(-comment, -project_name))[,colnames(mfd_db)]

mfd_db <- mfd_db %>%
  mutate(mfd_areatype        = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype      = ifelse(project_id == pid, "Water", mfd_sampletype),
         sampling_date       = ifelse(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table1,
            by = "fieldsample_barcode") %>%
  mutate(latitude            = ifelse(project_id == pid, correct_latitude, latitude),
         longitude           = ifelse(project_id == pid, correct_longitude, longitude),
         sitename            = ifelse(project_id == pid, correct_sitename, sitename),
         habitat_typenumber  = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         sampling_comment    = ifelse(project_id == pid, correct_comment, sampling_comment)) %>%
  select(-correct_latitude,
         -correct_longitude,
         -correct_sitename,
         -correct_habitat_typenumber,
         -correct_comment) %>%
  rbind(filled.table2) %>%
  mutate(habitat_typenumber  = ifelse(project_id == pid & habitat_typenumber == 2190, 3120, habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P06_3: Subterranean - Polluted
```{r P06_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P06_3"
pnm <- "Subterranean - Polluted"
des <- ""
ext <- ""
peo <- "Jeppe Lund Nielsen <jln@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Double check methods for inclusion in MfD."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P06_3_filled_table.xlsx"), sheet=1) %>%
  mutate(correct_habitat_typenumber = habitat_typenumber,
         correct_date = sampling_date) %>%
  select(fieldsample_barcode,
         correct_habitat_typenumber,
         correct_date)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         #habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "potential"), "6100", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(habitat_typenumber  = ifelse(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, correct_date, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_habitat_typenumber,
         -correct_date)

paragraph2 <- "Please double-check the dates beacuse they have a broad span."

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_1: Natural - Habitat Vision
```{r P08_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_1"
pnm <- "Natural - Habitat Vision"
des <- "One of our major projects with samples from >2500 natural habitats across Denmark. They are all classified after habitat type according to Natura 2000. Two overall goals: 1) look for diversity across habitats; 2) compare microorganisms in natural soil with related extensive fauna lists."
ext <- "Potentially - not inhouse currently."
peo <- "Erik Aude <eau@habitatvision.dk>; Lene Thomsen <lt@habitatvision.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Original habitat typenumbers provided as in: https://ecos.au.dk/fileadmin/ecos/Temasider/Raadgivning/TA-besigtigelse_af_naturarealer-105.pdf"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

rescued_samples <- c("\"D  5!5", "/F 0\"  2", "/FD!3 83", "/FD/3577", "/FDUP. 3", "2701", ">F&/4287", "M\"DO!Z X", "M\"DO45V5")

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P08_1_filled_table.xlsx")) %>%
  filter(!is.na(MFD_nr))
  
mapping.project <- filled.table %>%
  mutate(sample_name=`Relevé number`, sample_id=MFD_nr) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()

UTM32 <- CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m")

#UTM32n <- CRS("+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m +no_defs")
#WGS84 <- CRS("+proj=longlat +ellps=WGS84 +datum=WGS84") 

#utmcoor <- SpatialPoints(cbind(as.numeric(gsub("'|D", "", filled.table$Utm_x)),
#                               as.numeric(gsub("'|D", "", filled.table$Ytm_y))),
#                         proj4string=UTM32n)
lonlat <- geom(project(vect(cbind(as.numeric(gsub("'|D", "", filled.table$Utm_x)),
                      as.numeric(gsub("'|D", "", filled.table$Ytm_y))),
                      crs="+proj=utm +zone=32 +ellps=WGS84 +datum=WGS84 +units=m"),
                   "+proj=longlat +datum=WGS84"))[, c("x", "y")]

#longlatcoor <- spTransform(utmcoor, WGS84)

filled.table <- filled.table %>%
  mutate(fieldsample_barcode = MFD_nr,
         #correct_latitude = round(longlatcoor$coords.x2, 4),
         #correct_longitude = round(longlatcoor$coords.x1, 4),
         correct_longitude = round(lonlat[,1], 5),
         correct_latitude = round(lonlat[,2], 5),
         correct_sitename = Loknavn,
         correct_date = format(ymd(`Date (year/month/day)`), "%Y-%m-%d"),
         correct_habitat_typenumber = as.character(Naturnr)) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude,
         correct_sitename,
         correct_date,
         correct_habitat_typenumber)

haversine_distance <- function(lon1, lat1, lon2, lat2) {
  # Convert degrees to radians
  lon1_rad <- lon1 * pi / 180
  lat1_rad <- lat1 * pi / 180
  lon2_rad <- lon2 * pi / 180
  lat2_rad <- lat2 * pi / 180

  # Radius of the Earth in meters
  radius <- 6371000  # Approximate value

  # Haversine formula
  delta_lon <- lon2_rad - lon1_rad
  delta_lat <- lat2_rad - lat1_rad
  a <- sin(delta_lat/2)^2 + cos(lat1_rad) * cos(lat2_rad) * sin(delta_lon/2)^2
  c <- 2 * atan2(sqrt(a), sqrt(1 - a))
  distance <- radius * c

  return(distance)
}

tmp <- mfd_db %>%
  full_join(filled.table,
            by = "fieldsample_barcode") %>%
  filter(project_id==pid) %>%
  #mutate(d=sqrt(((correct_longitude-longitude)**2)+((correct_latitude-latitude)**2))*111.139) %>%
  mutate(d=haversine_distance(correct_longitude, correct_latitude, longitude, latitude)) %>%
  select(-received) %>%
  left_join(tmp_diff) %>%
  mutate(time_diff=ifelse(is.na(as.numeric(time_diff)), 0, time_diff)) %>%
  filter(!is.na(d),
         time_diff<500,
         d>100)

mfd_db <- mfd_db %>%
  full_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(project_id         = if_else(fieldsample_barcode %in% filled.table$fieldsample_barcode, pid, project_id),
         mfd_areatype       = ifelse(project_id == pid | fieldsample_barcode %in% rescued_samples, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid | fieldsample_barcode %in% rescued_samples, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid | fieldsample_barcode %in% rescued_samples, received, sampling_date),
         latitude           = if_else(project_id == pid & is.na(latitude), correct_latitude, latitude),
         longitude          = if_else(project_id == pid & is.na(longitude), correct_longitude, longitude),
         sitename           = if_else(project_id == pid & !is.na(correct_sitename), correct_sitename, sitename),
         sampling_date      = if_else(project_id == pid & !is.na(correct_date), correct_date, sampling_date),
         habitat_typenumber = if_else(project_id == pid & !is.na(correct_habitat_typenumber), correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_latitude,
         -correct_longitude,
         -correct_sitename,
         -correct_date,
         -correct_habitat_typenumber) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "1302", "1330", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4001", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4002", "4130", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "6201", "6210", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "6202", "6230", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "6203", "6120", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "6402", "6430", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "6403", "6431", habitat_typenumber), # now item
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7001", "7100", habitat_typenumber), # real: 7110, 7120
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7003", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7004", "7230", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7005", "7220", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7006", "6430", habitat_typenumber), # and 7210
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "7007", "7700", habitat_typenumber), # new item
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "9101", "91D0", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "9102", "91E0", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path)

paragraph2 <- "Due to a malfunctioning of the barcode recognistion caused by the iOS update, for few samples we retrieved nonsensical MFD barcodes and we cannot link the samples to the correct ones. Could you please double-check in your data and link the samples with the right barcodes?"

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_2: Natural - Other
```{r P08_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_2"
pnm <- "Natural - Other"
des <- "Comparing micro and macro in selected habitat types. Two master projects to be finished in 2021. Anne-Cathrine and Anne Mette later."
ext <- ""
peo <- "Dan Bruhn <db@bio.aau.dk>; Anne-Cathrine Storgaard Danielsen <acsd@agro.au.dk>; Lis Wollesen de Jonge <lis.w.de.jonge@agro.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Assuming dates are correct, but not sure."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P08_2_filled_table.xlsx")) %>%
  mutate(correct_sitename = sitename) %>%
  select(fieldsample_barcode, correct_sitename)
  
coord.correction <- read.xlsx(paste0(metadata.path, "/", pid, "/Correct coordinates.xlsx")) %>%
  mutate(correct_latitude = as.numeric(latitude),
         correct_longitude = as.numeric(longitude)) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(sitename           = if_else(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_sitename) %>%
  left_join(coord.correction, by = "fieldsample_barcode") %>%
  mutate(latitude           = if_else(project_id == pid & !is.na(correct_latitude), correct_latitude, latitude),
         longitude          = if_else(project_id == pid & !is.na(correct_longitude), correct_longitude, longitude)) %>%
  select(-correct_latitude,
         -correct_longitude)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_3: Natural - Other
```{r P08_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_3"
pnm <- "Natural - Other"
des <- "Other master projects with natural samples."
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Collage of multiple sub-projects; needs to be separated."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P08_3_filled_table.xlsx")) %>%
  mutate(correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode,
         correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(habitat_typenumber      = if_else((project_id == pid & !fieldsample_barcode %in% c("MFD03760", "MFD03795", "MFD03805",
                                                                                           "MFD03808", "MFD03846", "MFD03855",
                                                                                           "MFD03867", "MFD03871", "MFD03908",
                                                                                           "MFD03970", "MFD03973")), correct_habitat_typenumber, habitat_typenumber)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber)) %>% 
  add_ontology(ontology_path = ontology.path)  %>%
  select(-correct_habitat_typenumber)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_4: Biowide - Other
```{r P08_4, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_4"
pnm <- "Biowide - Other"
des <- "Samples collected in connection w P02_2, but will not be analyzed (replicate samples, mix of 3 sampling circles)"
ext <- ""
peo <- "Tobias Guldberg Frøslev < tobiasgf@sund.ku.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>; Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Samples collected in connection w P02_2, but will not be analyzed (replicate samples, mix of 3 sampling circles)."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_5: Natural - Sustain scapes
```{r P08_5, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_5"
pnm <- "Natural - Sustain scapes"
des <- "Collaboration about 750 samples for biodiversity studies in SustainScapes from primarily natural sites. AU works on the plant part, often in combination with Habitatvision. "
ext <- ""
peo <- "Signe Normand <signe.normand@bio.au.dk>; Urs Treier <urs.treier@bio.au.dk>; Bjarke Madsen <bjarke.madsen@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "10 samples are reallocared to P08_8: Natual - Sinks"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

correct_date <- readxl::read_excel(paste0(metadata.path, "/", pid, "/2022-07-08_mfd-metadata-sustainscapes.xlsx")) %>%
  select(fieldsample_barcode, tdate = sampling_date, tlat = latitude, tlon = longitude) %>%
  mutate(tdate = ymd_hms(tdate, truncated = 3) %>% format("%Y-%m-%d")) %>%
  filter(!(fieldsample_barcode %in% c("MFD06328", "MFD06329"))) # Two dual barcodes in the data

filled.table <- readr::read_delim(file = paste0(metadata.path, "/", pid, "/230524_SustainScapes_MFD_HabitatClasses_ForVibeke.txt"),
                                locale = readr::locale(encoding = "latin1"),
                                delim = ";",
                                col_names = T) %>%
  mutate(fieldsample_barcode = CORRECTED_barcode_fieldsample,
         sampling_date = ymd_hms(sampling_date, truncated = 3) %>% format("%Y-%m-%d"),
         correct_sampling_date = sampling_date,
         hab_code = as.character(hab_code),
         correct_habitat_typenumber = case_when(hab_code == "2" ~ NA,
                                                hab_code == "91" ~ "91E0",
                                                .default = hab_code),
         correct_latitude = latitude,
         correct_longitude = longitude,
         correct_sitename = hab_name,
         fieldsample_barcode = ifelse(fieldsample_barcode=="MFD6609", "MFD06609", fieldsample_barcode)) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude,
         correct_sampling_date,
         correct_habitat_typenumber,
         correct_sitename)

filled.table2 <- readxl::read_excel(paste0(metadata.path, "/P08_8/P08_8_frde_standard table.xlsx"))

filled.table1 <- (readr::read_delim(file = paste0(metadata.path, "/", pid, "/230524_SustainScapes_MFD_HabitatClasses_ForVibeke.txt"),
                                locale = readr::locale(encoding = "latin1"),
                                delim = ";",
                                col_names = T) %>%
  mutate(fieldsample_barcode = CORRECTED_barcode_fieldsample,
         fieldsample_barcode = ifelse(fieldsample_barcode=="MFD6609", "MFD06609", fieldsample_barcode)) %>%
  filter(!fieldsample_barcode%in%mfd_db$fieldsample_barcode, !fieldsample_barcode%in%filled.table2$fieldsample_barcode) %>%
  mutate(habitat_typenumber = hab_code,
         sampling_comment = hab_name,
         sample_type = type_sample,
         mfd_sampletype = "Soil",
         mfd_areatype = "Natural",
         mfd_hab1 = NA,
         mfd_hab2 = NA,
         mfd_hab3 = NA,
         habitat_type = NA,
         sitename = NA,
         project_id = pid,
         received = NA))[,colnames(mfd_db)] %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  left_join(correct_date, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         sampling_date      = ifelse(project_id == pid, tdate, sampling_date),
         latitude           = ifelse(project_id == pid, tlat, latitude),
         longitude          = ifelse(project_id == pid, tlon, longitude)) %>%
  select(-tdate, -tlat, -tlon) %>%
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(latitude = if_else((project_id==pid & is.na(latitude)), correct_latitude, latitude),
         longitude = if_else((project_id==pid & is.na(longitude)), correct_longitude, longitude),
         sampling_date = if_else((project_id==pid & is.na(sampling_date)), correct_sampling_date, sampling_date),
         sitename = if_else((project_id==pid & is.na(sitename)), correct_sitename, sitename),
         habitat_typenumber = if_else((project_id==pid & is.na(habitat_typenumber)), correct_habitat_typenumber, habitat_typenumber),
         mfd_sampletype = if_else((project_id == pid & habitat_typenumber %in% c("3110", "3130", "3100")), "Sediment", mfd_sampletype)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & mfd_sampletype == "Soil" & mfd_areatype == "Natural" & habitat_typenumber == "7007", "7700", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_latitude,
         -correct_longitude,
         -correct_sampling_date,
         -correct_sitename,
         -correct_habitat_typenumber) %>%
  rbind(filled.table1) %>%
  filter(!fieldsample_barcode %in% c("MFD10676", "MFD06329", "MFD10618")) # Duplicated samples as of 2023.06.07

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
rm(filled.table, filled.table1, filled.table2)
```

## P08_6: Natural - Untouched forest
```{r P08_6, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_6"
pnm <- " Natural - Untouched forest"
des <- "Samples from untouched forests (urørt skov). The samples are part of other studies on soil and biodiversity."
ext <- ""
peo <- "Mogens Humlegaard Greve <greve@agro.au.dk>; Lis Wollesen de Jonge <lis.w.de.jonge@agro.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "Check if the 107 samples and the new 250 are the same project (kvadratnet?) – see P23_1"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table1 <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P08_6_minimal_metadata_batch1&2.xlsx")) %>%
  separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
    select(-to_delete1, -to_delete2) %>%
  mutate(correct_date = ymd(sampling_date),
         correct_sitename = sitename,
         correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode,
         correct_date,
         correct_sitename,
         correct_habitat_typenumber)

filled.table2 <- (readxl::read_xlsx(paste0(metadata.path, "/", pid, "/P08_6_minimal_metadata_batch1&2.xlsx")) %>%
  filter(!fieldsample_barcode%in%mfd_db$fieldsample_barcode)%>%
  mutate(received = NA,
         project_id = pid,
         mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper)) %>%
    separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
    select(-to_delete1, -to_delete2))[,colnames(mfd_db)]

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         #habitat_typenumber = ifelse(project_id == pid, "9990", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  left_join(filled.table1,
                 by="fieldsample_barcode") %>%
  mutate(sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         sitename           = if_else(project_id == pid, correct_sitename, sitename),
         habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_date,
         -correct_sitename,
         -correct_habitat_typenumber) %>%
  rbind(filled.table2) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P08_7: Natural - ??
```{r P08_7, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_7"
pnm <- " Natural - ??"
des <- "The samples were collected from six different habitat types from the European Habitats Directive, comprising three types of grasslands (Calcareous grassland, Acidic grassland, and Sandy grassland) and three types of heathlands (Dry heath, Wet heath and Fixed dune). A total of 19 different locations (three to four locations representing each habitat type) were examined (Fig. 1A). Within each location, field surveys were conducted in three different sites (Fig. 1B) to cover variation within each location. (with the exception of five locations, where it was only possible to examine two sites, due to the limited area of the habitat type in question). Each site consisted of a circle with a radius of 5 meters (= 78.5 m2). A minimum distance of 20 meters was kept between sites. All field surveys were conducted in August and September 2020.
Prior to the collection of soil samples, botanical surveys were conducted by determining all species rooted within the site (complete species list, presence/absence) and positioning three pinpoint frames to analyse the frequency of the plant species within the area of the pinpoint frame. Soil samples (depth 0-16 cm) were collected at the centre of each pinpoint frame (the squares with crosses on Fig. 1C), after removal of vegetation and litter, and kept in a cool box. A subset of the three collected samples per site were pooled (resulting in the 55 soil samples in P08_2) but the rest was kept as single-samples (the 132 samples from this excel-file). Originally there were 165 single-samples, but some of them were discarded and therefore not sequenced or included in the metadata."
ext <- ""
peo <- "Anne-Cathrine Storgaard Danielsen <acsd@agro.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "GPS are wrong, but are related to P08_2 (individual samples from AC). Keep just on of the two?"

figure1.text <- "Figure 1: The geographical location of study areas (A), illustration of the sampling design within each location (B and C) and images of the habitat types (D) taken by the first author during field work. Each coloured circle on the map corresponds to one location. In five areas, the locations were located to close to show on the map, these locations are connected by a colourless circle and the connected locations were located in the centre of the colourless circle. Each coloured circle on (A), corresponds to the polygon seen on (B). The three circles within this polygon (B), illustrates the three sites in which field surveys were conducted. One of these sites is illustrated on (C) where the squares indicate pinpoint frames and x marks where soil samples were collected."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

data_correction <- readxl::read_excel(paste0(metadata.path, "/", pid, "/2021-12-22_P08_7-metadata.xlsx")) %>%
  mutate(latitude = as.numeric(latitude),
         longitude = as.numeric(longitude)) %>%
  select(fieldsample_barcode,
         tdate = received,
         thab = habitat_typenumber,
         correct_latitude = latitude,
         correct_longitude = longitude,
         correct_sitename = sitename) %>%
  mutate(tdate = ymd_hms(tdate, truncated = 3) %>% format("%Y-%m-%d"))

mapping.project <- readxl::read_excel(paste0(metadata.path, "/", pid, "/2021-12-22_P08_7-metadata.xlsx"),
                                      sheet = "AC_extended") %>%
  filter(!is.na(miljoeportal)) %>%
  mutate(sample_name=miljoeportal, sample_id=barcode_fieldsample) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()

mfd_db <- mfd_db %>%
  left_join(data_correction, by = "fieldsample_barcode") %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, thab, habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, tdate, sampling_date)) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4010", "4110", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Soil" & habitat_typenumber == "4030", "4130", habitat_typenumber),
         latitude           = if_else(project_id == pid, correct_latitude, latitude),
         longitude          = if_else(project_id == pid, correct_longitude, longitude),
         sitename           = if_else(project_id == pid, correct_sitename, sitename)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-tdate, -thab, -correct_latitude, -correct_longitude, -correct_sitename)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")

#![`r figure1.text`](`r paste0(metadata.path, "/P08_7/sampling_locations.png")`)
```

## P08_8: Natural - SINKS
```{r P08_8, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P08_8"
pnm <- " Natural - SINKS"
des <- ""
ext <- ""
peo <- "Signe Normand <signe.normand@bio.au.dk>; Urs Treier <urs.treier@bio.au.dk>; Bjarke Madsen <bjarke.madsen@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "Samples reallocared from P08_5: Natual - Sustain scapes"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- (readxl::read_excel(paste0(metadata.path, "/P08_8/P08_8_frde_standard table.xlsx")) %>%
                   mutate(received = NA) %>%
  select(-project_name))[,colnames(mfd_db)]

mfd_db <- mfd_db %>%
  rbind(filled.table) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P09_1: Lakes - Habitat Vision
```{r P09_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P09_1"
des <- "Samples from sediments from many lakes across DK."
pnm <- "Lakes - Habitat Vision"
ext <- ""
peo <- "Erik Aude <eau@habitatvision.dk>; Lene Thomsen <lt@habitatvision.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Lakes sampled by Habitat Vision."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P09_1_filled_table.xlsx")) %>%
  filter(!is.na(MFD_nr))

mapping.project <- filled.table %>%
  mutate(sample_name=`Relevé number`, sample_id=MFD_nr) %>%
  select(sample_name, sample_id) %>%
  as.data.frame()

utmcoor <- SpatialPoints(cbind(filled.table$Utm_x, filled.table$Ytm_y), proj4string=CRS("+proj=utm +zone=32"))
longlatcoor <- spTransform(utmcoor,CRS("+proj=longlat"))

filled.table <- filled.table %>%
  mutate(fieldsample_barcode = MFD_nr,
         correct_latitude = longlatcoor$coords.x2,
         correct_longitude = longlatcoor$coords.x1,
         correct_sitename = Loknavn,
         correct_date = format(ymd(`Date (year/month/day)`), "%Y-%m-%d"),
         correct_habitat_typenumber = as.character(Naturnr)) %>%
  select(fieldsample_barcode,
         correct_latitude,
         correct_longitude,
         correct_sitename,
         correct_date,
         correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  full_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(project_id         = if_else(fieldsample_barcode %in% filled.table$fieldsample_barcode, pid, project_id),
         latitude           = if_else(project_id == pid, correct_latitude, latitude),
         longitude          = if_else(project_id == pid, correct_longitude, longitude),
         sitename           = if_else(project_id == pid, correct_sitename, sitename),
         sampling_date      = if_else(project_id == pid, correct_date, sampling_date),
         habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype)) %>%
  select(-correct_latitude,
         -correct_longitude,
         -correct_sitename,
         -correct_date,
         -correct_habitat_typenumber) %>%
  mutate(habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Sediment" & habitat_typenumber == "3103", "3150", habitat_typenumber),
         habitat_typenumber = if_else(project_id == pid & mfd_areatype == "Natural" & mfd_sampletype == "Sediment" & habitat_typenumber == "3104", "3160", habitat_typenumber)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P09_2: Lakes - Ponderfull
```{r P09_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P09_2"
pnm <- "Lakes - Ponderfull"
des <- "Arhus Uni has a project \"Ponderful\" where they sample from 30 small ponds (vandhul) – to look for cable bacteria"
ext <- ""
peo <- "Ian Marshall <ianpgm@bio.au.dk>; Kasper Urup Kjeldsen <kasperuk@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P09_3: Lakes -Sediments SDU
```{r P09_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P09_3"
pnm <- "Lakes -Sediments SDU"
des <- "Sediment samples from 10 lakes, most in 2-3 depths (0-20 cm). Are used for chemistry, including polyp, analyses at SDU."
ext <- ""
peo <- "Kasper Reitzel <reitzel@biology.sdu.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Marta has made amplicon analyses on the samples. Metadata exist."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P09_3_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
         mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper),
         sampling_date = sampling_date  %>% format("%Y-%m-%d"),
    latitude = as.numeric(latitude),
    longitude = as.numeric(longitude),
    received = NA) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P09_4: Lakes - Sediments Grundfos
```{r P09_4, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P09_4"
pnm <- "Lakes - Sediments Grundfos"
des <- "Sediment samples from Grundfos lake (Ormstrup Gods) by SDU. Samples from 5 stations in the lake – and from 5 different sediment depth layers (spanning 0-20 cm). Sampling sets from August 2022 and February 2023. There are 50 samples in total"
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Responsible: Poul Toft Frederiksen, no mail found."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P09_4_frde_standard table.xlsx")) %>%
  mutate(sampling_date = sampling_date  %>% format("%Y-%m-%d"),
         project_id = pid,
         mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper)) %>%
  separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
  select(-to_delete1, -to_delete2) %>%
  add_ontology(ontology_path = ontology.path) %>%
  #select(-project_name) %>%
  mutate(received = NA)
  
mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])
  #full_join(filled.table,
  #          by = "fieldsample_barcode")

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P10_1: Streams - Habitat Vision
```{r P10_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P10_1"
pnm <- "Streams - Habitat Vision"
des <- "Sediment samples streams. Different classification of stream condition than for project P10_2."
ext <- ""
peo <- "Erik Aude <eau@habitatvision.dk>; Lene Thomsen <lt@habitatvision.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P10_2: Streams - DVFI
```{r P10_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P10_2"
pnm <- "Streams - DVFI"
des <- "Sediment samples streams. From DVFI stations, and collected in spring to be comparable to DVFI results. A lot of very specific additional data collected in field Sampling only in spring."
ext <- ""
peo <- "Jeppe Lund Nielsen <jln@bio.aau.dk>; Nadieh de Jonge <ndj@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P10_2_filled_table.xlsx")) %>%
  mutate(correct_sitename=sitename) %>%
  select(fieldsample_barcode, correct_sitename)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "3200", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table, by = "fieldsample_barcode") %>%
  mutate(sitename           = if_else(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_sitename) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P10_3: Streams - students
```{r P10_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P10_3"
pnm <- "Streams - students"
des <- "Other stream samples collected by MFD/students (without additional classification needed for the big projects)."
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P11_1: Coasts - students
```{r P11_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P11_1"
pnm <- "Coasts - students"
des <- "Investigate microorganisms at different salinity (Kattegat vs Skagerak). NB: extraction of sand sediment did not yield sufficient DNA – instead based on Fastprep DNA extraction (not comparable with other projects)"
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)
mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1100", habitat_typenumber),         
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P11_2: Coasts and Fjords - Philip
```{r P11_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P11_2"
pnm <- "Coasts and Fjords - Philip"
des <- "Other stream samples collected by MFD/students (without additional classification needed for the big projects). 160 DNA extractions of sediment samples collected in the COAST_SEQUENCE project at AU. See also https://bio.au.dk/en/research/research-areas/biodiversity/marine-life-around-denmark"
ext <- ""
peo <- "Philip Francis Thomsen <pfthomsen@bio.au.dk>; Martin Johannesen Klepke <martinjk@bio.au.dk>; Marie Vestergård <mv@bio.au.dk"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Samples MFD10339 and MFD10290 had labels that were impossible to distinguish, therefore they have been removed from the analysis."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

to_remove <- c("MFD10339", "MFD10290")

mapping.project <- readxl::read_xlsx(paste0(metadata.path, "/", pid, "/2023-04-25_P11_2-mapping.xlsx")) %>%
  filter(!is.na(MFD)) %>%
  mutate(sample_name = `collab Sample number`,
         sample_id = MFD) %>%
  select(sample_name, sample_id) %>%
  as.data.frame() %>%
  filter(!sample_id %in% to_remove)

filled.table <- read.xlsx(paste0(metadata.path, "/", pid, "/P11_2_filled_table.xlsx"),
                          sheet = 1,
                          detectDates = T) %>%
  mutate(correct_date = as.character(sampling_date),
         correct_latitude = as.numeric(latitude),
         correct_longitude = as.numeric(longitude),
         correct_habitat_typenumber = habitat_typenumber,
         correct_sitename = sitename) %>%
  select(fieldsample_barcode,
         correct_date,
         correct_latitude,
         correct_longitude,
         correct_habitat_typenumber,
         correct_sitename) %>%
  filter(!fieldsample_barcode %in% to_remove)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "coast" & str_detect(sampling_comment, "sandy"),"1101", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "coast" & str_detect(sampling_comment, "rocky"),"1102", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "coast" & str_detect(sampling_comment, "eelgrass"),"1103", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "fjords" & str_detect(sampling_comment, "sandy"),"1301", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "fjords" & str_detect(sampling_comment, "rocky"),"1302", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "fjords" & str_detect(sampling_comment, "eelgrass"),"1303", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(sampling_date      = if_else(project_id == pid, correct_date, sampling_date),
         latitude           = if_else(project_id == pid, correct_latitude, latitude),
         longitude          = if_else(project_id == pid, correct_longitude, longitude),
         habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         sitename           = if_else(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_date,
         -correct_latitude,
         -correct_longitude,
         -correct_habitat_typenumber,
         -correct_sitename) %>%
  filter(!fieldsample_barcode %in% to_remove) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P11_3: Fjords - Niels
```{r P11_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P11_3"
pnm <- "Fjords - Niels"
des <- "Investigate microorganisms in water and sediment from Danish Fjords."
ext <- ""
peo <- "Niels Iversen <ni@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Add method on how samples were taken."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, str_to_title(sample_type), mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1300", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P12_1: Deep Sea - Aarhus
```{r P12_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P12_1"
pnm <- "Deep Sea - Aarhus"
des <- "Kasper: 30 samples, where extensive geochemical metadata is awailable, as well as 16S rRNA amplicon seq and qPCR. Samples from the sulfate-methane transition zone. MFD contribute possibility of making FISH probes via full operon seq. Ian: deep sediments from IODP togt in 2013, where focus have been on hunting for cablebacteria. Initially, we got DNA from 60 sediment samples from 6 sites. However, the sequencing did not work. New project: Samples from 5 stations and 3-4 depths are sampled (extracted DNA delivered February 2023). Extraction procedure to follow. The samples are from the sulfate-methane transition zone, primarily to look for ANME. The project also contains a part with deep sequencing (Caitlin) to obtain ANME MAGs. "
ext <- ""
peo <- "Kasper Urup Kjeldsen <kasperuk@bio.au.dk>; Ian Marshall <ianpgm@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Caitlin Margaret Singleton <cms@bio.aau.dk>"
com <- "The ANME part is in a separate project with deep sequencing (Caitlin). Samples received as DNA from phenol-chloroform extraction – worked really bad."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_xlsx(paste0(metadata.path, "/", pid, "/P12_1_minimal_metadata_updated.xlsx"))

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "bornholm"), "1210", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "kattegat"), "1220", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "skagerrak"), "1230", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "lillebælt"), "1110", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "aarhus"), "1110", habitat_typenumber),         
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  rbind(filled.table[!filled.table$fieldsample_barcode%in%mfd_db$fieldsample_barcode,colnames(mfd_db)]) %>%
  filter(!(project_id==pid & grepl("negative control", sampling_comment))) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P12_2: Deep Sea - Aarhus
```{r P12_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P12_2"
pnm <- "Deep Sea - Aarhus"
des <- "Deep sediments from IODP cruise in 2013, where focus has been on hunting for cable bacteria"
ext <- ""
peo <- "Kasper Urup Kjeldsen <kasperuk@bio.au.dk>; Ian Marshall <ianpgm@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "bornholm"), "1210", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "anholt"), "1220", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sitename, "lille"), "1110", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P12_3: Fjords - Aarhus
```{r P12_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P12_3"
pnm <- "Fjords - Aarhus"
des <- "Seabed Mariager fjord"
ext <- ""
peo <- "Ian Marshall <ianpgm@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Fill in description with methods."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1310", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P12_4: Deep Sea - SDU
```{r P12_4, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P12_4"
pnm <- "Deep Sea - SDU"
des <- "Samples from the deep trenches (> 5000 m depths). Compare microbial communities in different trenches and different depth."
ext <- ""
peo <- "Ronnie Glud <rnglud@biology.sdu.dk>, Clemens Schauberger <schauberger@biology.sdu.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Ronnie Glud <rnglud@biology.sdu.dk>"
com <- "Not included in MFD."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Sediment", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "kamchatska"),"1420", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & str_detect(sampling_comment, "atacama"),"1410", habitat_typenumber),       
         latitude           = ifelse(project_id == pid, NA, latitude),
         longitude          = ifelse(project_id == pid, NA, longitude),          
         sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P12_4_filled_table.xlsx")) %>%
  mutate(correct_date = sampling_date,
         correct_latitude = latitude,
         correct_longitude = longitude,
         correct_sitename = sitename) %>%
  select(fieldsample_barcode,
         correct_date,
         correct_latitude,
         correct_longitude,
         correct_sitename)

mfd_db <- left_join(mfd_db, filled.table,
                 by="fieldsample_barcode") %>%
  mutate(sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         latitude           = if_else(project_id == pid, correct_latitude, latitude),
         longitude          = if_else(project_id == pid, correct_longitude, longitude),
         sitename           = if_else(project_id == pid, correct_sitename, sitename)) %>%
  select(-correct_date,
         -correct_latitude,
         -correct_longitude,
         -correct_sitename)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P12_5: Sea sediments – Baltic sea
```{r P12_5, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P12_5"
pnm <- "Sea sediments – Baltic sea"
des <- "Sediment samples from Baltic Sea.  Sampled by “Miljøstyrelsen”, and they have extensive metadata."
ext <- ""
peo <- "Sanne Kjellerup <Sanne.Kjellerup@wsp.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "The samples arrived in December 2023. Initially created as P22_1."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P12_5_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
    sampling_date = sampling_date  %>% format("%Y-%m-%d"),
    mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper),
    received = NA,
    longitude = as.numeric(longitude),
    latitude = as.numeric(latitude)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P13_1: Urban - Influent Wastewater
```{r P13_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P13_1"
pnm <- "Urban - Influent Wastewater"
des <- "Influent wastewater samples. Compare wastewater inlet to sludge from receiving WWTP’s. Samples have already been analyzed by 16S rRNA amplicon seq. Influent wastewater samples. Compare wastewater inlet to activated sludge from corresponding WWTPs. Samples have already been analyzed by 16S rRNA amplicon seq. Most samples are pooled from 10 day samples (24h flow proportional) over 3 weeks. The goal is to look for diversity across DK, importance of immigration, any geographical pattern. "
ext <- ""
peo <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Marta Anna Nierychlo <mni@bio.aau.dk>; Giulia Dottorini <gd@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "We should add additional 13 MiDAS plants. Add extraction method + pooling, etc. to description."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1210", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P13_2: Urban - AS Wastewater
```{r P13_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P13_2"
pnm <- "Urban - AS Wastewater"
des <- "Corresponding activated sludge samples to influent samples P13_1. Most samples are pooled from 10 grab samples over 3 weeks. The goal is to look for diversity across DK, importance of immigration, any geographical pattern. Samples have already been analyzed by 16S rRNA amplicon seq."
ext <- ""
peo <- "Per Halkjær Nielsen <phn@bio.aau.dk>; Marta Anna Nierychlo <mni@bio.aau.dk>; Giulia Dottorini <gd@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "We should add additional 14 MiDAS plants. Add extraction method etc. to description."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_xlsx(paste0(metadata.path, "/", pid, "/P13_2_table_to_fill_Wextrasamples2.xlsx")) %>%
  filter(!fieldsample_barcode%in%mfd_db$fieldsample_barcode) %>%
  mutate(habitat_typenumber = ifelse(habitat_typenumber == "2210", "1210", habitat_typenumber),
         habitat_typenumber = ifelse(habitat_typenumber == "2211", "1211", habitat_typenumber),
         longitude = as.numeric(longitude),
         latitude = as.numeric(latitude),
         received = NA,
         sampling_comment = NA,
         project_id = pid,
         sampling_date = sampling_date %>% format("%Y-%m-%d"))  %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1211", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  rbind(filled.table[!filled.table$fieldsample_barcode%in%mfd_db$fieldsample_barcode,colnames(mfd_db)]) %>%
  #filter(!(project_id==pid & grepl("negative control", sampling_comment))) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P13_3: Urban - Biogas
```{r P13_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P13_3"
pnm <- "Urban - Biogas"
des <- "Wastewater treatment plant samples."
ext <- ""
peo <- "Caitlin Margaret Singleton <cms@bio.aau.dk>; Chenjing Jiang <chj@bio.aau.dk>"
res <- "Caitlin Margaret Singleton <cms@bio.aau.dk>; Chenjing Jiang <chj@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- (readxl::read_xlsx(paste0(metadata.path, "/", pid, "/P13_3_frde_standard table1_chj_20230807.xlsx")) %>%
  filter(!fieldsample_barcode%in%mfd_db$fieldsample_barcode)%>%
  mutate(received = NA,
         project_id = pid,
         mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper)) %>%
    separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
    select(-to_delete1, -to_delete2) %>%
    mutate(latitude = as.numeric(latitude),
           longitude = as.numeric(longitude)) %>%
    select(-project_name))[,colnames(mfd_db)]

mfd_db <- mfd_db %>%
  rbind(filled.table) %>%
  mutate(habitat_typenumber = ifelse((project_id == pid & habitat_typenumber == "2111"), "1411", habitat_typenumber),
         habitat_typenumber = ifelse((project_id == pid & habitat_typenumber == "2112"), "1412", habitat_typenumber),
         habitat_typenumber = ifelse((project_id == pid & habitat_typenumber == "2119"), "1419", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  #rbind(filled.table[!filled.table$fieldsample_barcode%in%mfd_db$fieldsample_barcode,colnames(mfd_db)]) %>%
  #filter(!(project_id==pid & grepl("negative control", sampling_comment))) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P16_1: Urban - Drinking water
```{r P16_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P16_1"
pnm <- "Urban - Drinking water"
des <- "Samples collected from water plants. Filtered onto filters."
ext <- ""
peo <- "Peter Roslev <pr@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse((project_id == pid & habitat_typenumber == "3000"), "3300", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, received, sampling_date)) %>% # Dates are correct
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P16_2: Urban - Drinking water
```{r P16_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P16_2"
pnm <- "Urban - Drinking water"
des <- "Samples collected from water treatment plants. Filtered onto filters."
ext <- ""
peo <- "Horsens Torben Lund Skovhus <tols@via.dk>; Ditte Andreasen Søborg <DANS@via.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Very varying results for prep and sequencing. Sampling dates exist but have to be fetched."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "3000", "3300", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P16_2_filled_table.xlsx")) %>%
  mutate(correct_date = format(sampling_date, "%Y-%m-%d")) %>%
  select(fieldsample_barcode, correct_date)

mfd_db <- left_join(mfd_db, filled.table,
                 by = "fieldsample_barcode") %>%
  mutate(sampling_date      = ifelse(project_id == pid, correct_date, sampling_date)) %>%
  select(-correct_date)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P16_3: Urban - Drinking water
```{r P16_3, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P16_3"
pnm <- "Urban - Drinking water"
des <- "Sjælland: Samples collected from water treatment plants (water works). Filtered onto filters."
ext <- ""
peo <- "Barth F. Smets <bfsm@dtu.dk>; Marlene Mark Jensen <mmaj@env.dtu.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_typenumber == "3000", "3300", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P16_4: Urban – Groundwater
```{r P16_4, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P16_4"
pnm <- "Urban – Groundwater"
des <- "Sjælland: Samples collected from water treatment plants (water works). Filtered onto filters."
ext <- ""
peo <- "Jeppe Lund Nielsen <jln@bio.aau.dk>; Nadieh de Jonge <ndj@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_xlsx(paste0(metadata.path, "/", pid, "/P16_4_frde_standard table.xlsx")) %>%
  mutate(habitat_typenumber = as.character(habitat_typenumber),
         habitat_typenumber = case_when(habitat_typenumber == "2000" ~ "3110",
                                        habitat_typenumber == "2110" ~ "3130",
                                        habitat_typenumber == "2190" ~ "3140",
                                        .default = habitat_typenumber),
         correct_habitat_typenumber = habitat_typenumber) %>%
  select(fieldsample_barcode, correct_habitat_typenumber)

mfd_db <- mfd_db %>%
  left_join(filled.table,
            by = "fieldsample_barcode") %>%
  mutate(habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber),
         mfd_sampletype     = if_else(project_id == pid, "Water", mfd_sampletype),
         mfd_areatype       = if_else(project_id == pid, "Subterranean", mfd_areatype),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path) %>%
  select(-correct_habitat_typenumber)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P17_1: Urban - Soil
```{r P17_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P17_1"
pnm <- "Urban - Soil"
des <- "Soil collected from Danish cities across Denmark, up to 3 soil samples from same city covering different areatypes, e.g. parks, industrial, side of the road etc (same cities are also collected urban ponds P05_2 and rainwater P05_1)."
ext <- ""
peo <- ""
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "People Erika Dvarionaite and Amalie Berg-Mortensen (their emails need to be retrieved)."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Urban", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "city_parks", "6410", habitat_typenumber),
         habitat_typenumber = ifelse(project_id == pid & habitat_type == "city_other", "6420", habitat_typenumber),         
         sampling_date      = if_else(project_id == pid, received, sampling_date),
         sampling_date      = ifelse(fieldsample_barcode=="MFD10966", "2022-05-23", sampling_date),
         sampling_date      = ifelse(fieldsample_barcode=="MFD02265", "2020-07-08", sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P18_1: Urban - Harbour
```{r P18_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P18_1"
pnm <- "Urban - Harbour"
des <- "Harbors around DK, surfaces are scraped of stone, tree. To be handled as sediment. To be used for other investigations at DTU. Are interested in BGC-producers, such as Phaeobacter or Pseudoalteromonas. Student helpers Majbrit and Camilla."
ext <- "Salinity, temperature etc."
peo <- "Lone Gram, <gram@bio.dtu.dk>; Peter Bing Svendsen, <pesven@dtu.dk>; Morten Dencker Schostag <mdesc@dtu.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "Should be an <urban> project instead?"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Natural", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Water", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "1110", habitat_typenumber),
         sampling_date      = if_else(project_id == pid, received, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P18_2: MfD Harbour
```{r P18_2, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P18_2"
pnm <- "MfD Harbour"
des <- "Harbours around DK, surfaces are scraped of stone, tree. To be used for other investigations at DTU. Are interested in BCG-producers, such as Phaeobacter or Pseudoalteromonas.  Samples collected from east and west “sides” of Jylland. Collected by our student helpers Majbrit and Camilla."
ext <- "Salinity, temperature etc."
peo <- "Lone Gram, <gram@bio.dtu.dk>; Peter Bing Svendsen, <pesven@dtu.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P18_2_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
    sampling_date = sampling_date  %>% format("%Y-%m-%d"),
    mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper),
    received = NA) %>%
  separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
  select(-to_delete1, -to_delete2) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P19_1: Urban - Archaeological
```{r P19_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P19_1"
pnm <- "Urban - Archaeological"
des <- "Samples from excavations (1000 year old)"
ext <- ""
peo <- "Torsten Nygård Kristensen <tnk@bio.aau.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

mfd_db <- mfd_db %>%
  mutate(mfd_areatype       = ifelse(project_id == pid, "Subterranean", mfd_areatype),
         mfd_sampletype     = ifelse(project_id == pid, "Soil", mfd_sampletype),
         habitat_typenumber = ifelse(project_id == pid, "6200", habitat_typenumber),
         sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  add_ontology(ontology_path = ontology.path)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P19_1_filled_table.xlsx")) %>%
  mutate(correct_date = format(sampling_date, "%Y-%m-%d")) %>%
  select(fieldsample_barcode, correct_date)

mfd_db <- left_join(mfd_db, filled.table,
                 by="fieldsample_barcode") %>%
  mutate(sampling_date      = ifelse(project_id == pid, correct_date, sampling_date)) %>%
  select(-correct_date)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P20_1: Urban - Landfills
```{r P20_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P20_1"
pnm <- "Urban - Landfills"
des <- "6 samples from AV Miljø landfill at Avedøre Holme in Hvidovre, Copenhagen. 4 soil samples and 2 leachate samples. Furthermore 22 laboratory incubations with different plastic polymers. The 2 leachate samples were acquired as DNA extracted with FastDNA™ SPIN Kit for Soil."
ext <- ""
peo <- "Cristiano Varrone <cva@bio.aau.dk>; Passanun Lomwongsopon <paslom@bio.aau.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- "Incubations doesn’t fit the remaining MFD samples."

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P20_1_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
         mfd_sampletype = str_replace(mfd_sampletype, "^\\w{1}", toupper),
         mfd_areatype = str_replace(mfd_areatype, "^\\w{1}", toupper),
         sampling_date = sampling_date  %>% format("%Y-%m-%d"),
         habitat_typenumber = case_when(habitat_typenumber == "wa_u_6310" ~ "wa_u_1310",
                                   habitat_typenumber == "se_u_6310" ~ "se_u_2110",
                                   .default = habitat_typenumber), 
    latitude = as.numeric(latitude),
    longitude = as.numeric(longitude),
    received = NA) %>%
  separate(habitat_typenumber, into = c("to_delete1", "to_delete2", "habitat_typenumber")) %>%
  select(-to_delete1, -to_delete2) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P21_1: Freshwater Seep - AU
```{r P21_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P21_1"
pnm <- "Freshwater Seep - AU"
des <- "Sediment samples from freshwater seep (kildevæld), Salten Skov. Has special Fe-hydroxid precipitates which are rare in Denmark. Is used as Mars-similar location. Paper: https://www.tandfonline.com/doi/abs/10.1080/01490450903232165"
ext <- ""
peo <- "Kai Finster <kai.finster@bio.au.dk>"
res <- "Per Halkjær Nielsen <phn@bio.aau.dk>"
com <- "5 samples. Metadata, including photos exist"

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P21_1_frde_standard table.xlsx")) %>%
  mutate(project_id = pid,
         mfd_sampletype = "Soil",
         mfd_areatype = "Natural",
         sampling_date = dmy(sampling_date)  %>% format("%Y-%m-%d"),
         latitude = as.numeric(latitude),
         longitude = as.numeric(longitude),
         received = NA) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db <- mfd_db %>%
  rbind(filled.table[,colnames(mfd_db)])

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

## P25_1: Other - LHP
```{r P25_1, echo=F, message=F, warning=F, error=F, results='asis'}
pid <- "P25_1"
pnm <- "Other - LHP"
des <- "Miscellaneous samples from LHP. Samples are from 2008 and was thawed and put in the refrigerator 2 months before subdivision. Samples are either from chalk mine, salt vats or bogs."
ext <- ""
peo <- "Lars Haastrup Pedersen <lhp@bio.aau.dk>"
res <- "Thomas Bygh Nymann Jensen <tbnj@bio.aau.dk>"
com <- ""

metadata.project <- paste0(pid, "/", pid,  "_metadata_summary.csv")

mfd_projects[mfd_projects$project_id == pid,] <- list(pid, pnm, des, ext, peo, res, com, metadata.project)

filled.table <- readxl::read_excel(paste0(metadata.path, "/", pid, "/P25_1_frde_standard table.xlsx")) %>%
  mutate(correct_mfd_sampletype = case_when(habitat_type == "natural_soil" ~ "Soil",
                                   habitat_type == "built_environment" ~ "Sediment",
                                   .default = mfd_sampletype), 
         correct_mfd_areatype = case_when(habitat_type == "natural_soil" ~ "Natural",
                                   habitat_type == "built_environment" ~ "Urban",
                                   .default = mfd_areatype),
         habitat_typenumber = as.character(habitat_typenumber),
         habitat_typenumber = case_when(habitat_typenumber == "9110" ~ "2120",
                                        habitat_typenumber == "9120" ~ "2130",
                                        .default = habitat_typenumber),
         correct_habitat_typenumber = as.character(habitat_typenumber),
         correct_date = dmy(sampling_date)) %>%
  select(fieldsample_barcode, correct_mfd_sampletype, correct_mfd_areatype, correct_habitat_typenumber, correct_date)

mfd_db <- mfd_db %>%
  #mutate(sampling_date      = ifelse(project_id == pid, NA, sampling_date)) %>%
  left_join(filled.table,
                 by = "fieldsample_barcode") %>%
  mutate(sampling_date      = ifelse(project_id == pid, correct_date, sampling_date),
         mfd_sampletype     = if_else(project_id == pid, correct_mfd_sampletype, mfd_sampletype),
         mfd_areatype       = if_else(project_id == pid, correct_mfd_areatype, mfd_areatype),
         habitat_typenumber = if_else(project_id == pid, correct_habitat_typenumber, habitat_typenumber)) %>%
  select(-correct_date, -correct_mfd_sampletype, -correct_mfd_areatype, -correct_habitat_typenumber) %>%
  add_ontology(ontology_path = ontology.path)

mfd_db_subproject <- mfd_subset(project = pid, data_table = mfd_db)
N <- nrow(mfd_db_subproject)

res <- knitr::knit_child(paste0(wd, "/subproject_template.Rmd"), envir = globalenv(), quiet = T)
cat(res, sep = "\t")
```

# General outputs

## Output checks

This is the distribution of sample names in the curated metadata:

### Sample duplication check
```{r, warning=T, echo=eval_debug, eval=T}
table(table(mfd_db$fieldsample_barcode))
```

### Retain samples with project numbers
```{r, warning=T, echo=eval_debug, eval=T}
mfd_db <- mfd_db %>%
  filter(!is.na(project_id))
table(table(mfd_db$fieldsample_barcode))
```

## Write output

### Write project map
```{r, warning=T, echo=eval_debug, eval=T}
print_internal_map(data_table = mfd_projects, fname = paste0(reformat.metadata.path, "/mfd_projects.xlsx"))
```

### Write mfd db
```{r, warning=T, echo=eval_debug, eval=T}
writexl::write_xlsx(mfd_db, path = paste0(reformat.metadata.path, "/mfd_db.xlsx"))
```

## Write releases

### Write project map
```{r, warning=T, echo=eval_debug, eval=eval_release}
print_internal_map(data_table = mfd_projects, fname = paste0(releases.path, "/", Sys.Date(), "_mfd_projects.xlsx"))
```

### Write mfd db
```{r, warning=T, echo=eval_debug, eval=eval_release}
writexl::write_xlsx(mfd_db[,release_indices], path = paste0(releases.path, "/", Sys.Date(), "_mfd_db.xlsx"))
```